modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/coordination/rdbms/RDBMSAgent.java

```diff
@@ -163,7 +163,9 @@ public void addMessageId(String queueName, long messageId) throws AndesException
 		try {
 			andesContextStore.addMessageId(queueName, messageId);
 		} catch (AndesDataIntegrityViolationException ignore) {
-			//Primary key violation exception ignored
+			//Same message id can be added to list when slots are overlapped. In RDBMS slot store
+			//composite primary key of queue name and message id have been used to avoid duplicates.
```

Comment:
from what I know there a vendor specific syntax ( - sql ) which will help us prevent this. Obviously if we do that, we will be writing stuff for each db vendor.